include/fdreadoutlibs/wibeth/tpg/ProcessNaiveRS.hpp

```diff
@@ -95,11 +95,20 @@ process_window_naive_RS(ProcessingInfo<NREGISTERS>& info)
       // Naive: RS = (R * RS) + std::abs(sample)/scale
       // RS = RS * RS_memory_factor + abs(sample) / RS_scale_factor;
 
+      /*
       int16_t first_part = (int16_t)(RS * RS_memory_factor);
       int16_t second_part = (int16_t)(std::abs(sample) * RS_scale_factor);
 
       first_part = naive_avx2_div(first_part, (int16_t)10);
       second_part = naive_avx2_div(second_part, (int16_t)10);
+      RS = (int16_t)(first_part + second_part);
+      */
+
+      int16_t first_part = naive_avx2_div(RS, (int16_t)10);
+      first_part = (int16_t)(first_part * RS_memory_factor);
+      int16_t second_part = naive_avx2_div(std::abs(sample), (int16_t)10);
+      second_part = (int16_t)(second_part * RS_scale_factor);
```

Comment:
Since we want to do the saturated addition in the AVX2 version, a similar case would be needed here. This can be done with something like the following:
```cpp
int32_t overflow_part = first_part + second_part;
RS = std::min(overflow_part, INT16_MAX);
```